src/field/generated_fieldops.cxx

```diff
@@ -0,0 +1,1668 @@
+// This file is autogenerated - see gen_fieldops.py
+#include <bout/mesh.hxx>
+#include <field2d.hxx>
+#include <field3d.hxx>
+#include <globals.hxx>
+#include <interpolation.hxx>
+
+// Do the actual multiplication of Field3D and Field3D
+void autogen_Field3D_Field3D_Field3D_multiplication(BoutReal *__restrict__ result,
+                                                    const BoutReal *__restrict__ lhs,
+                                                    const BoutReal *__restrict__ rhs,
+                                                    int len) {
+
+  for (int i = 0; i < len; ++i) {
+    result[i] = lhs[i] * rhs[i];
+  }
+}
+
+// Provide the C++ wrapper for multiplication of Field3D and Field3D
+Field3D operator*(const Field3D &lhs, const Field3D &rhs) {
+  Indices i{0, 0, 0};
+  Mesh *localmesh = lhs.getMesh();
+
+  ASSERT1(localmesh == rhs.getMesh());
+
+  Field3D result(localmesh);
+  result.allocate();
+  checkData(lhs);
+  checkData(rhs);
+
+  autogen_Field3D_Field3D_Field3D_multiplication(&result[i], &lhs[i], &rhs[i],
+                                                 localmesh->LocalNx * localmesh->LocalNy *
+                                                     localmesh->LocalNz);
+
+#if CHECK > 0
+  if (lhs.getLocation() != rhs.getLocation()) {
+    throw BoutException("Error in multiplication: fields at different locations. lhs is "
+                        "at %s, rhs is at %s!",
+                        strLocation(lhs.getLocation()), strLocation(rhs.getLocation()));
+  }
+#endif
+
+  result.setLocation(rhs.getLocation());
+
+  checkData(result);
+  return result;
+}
+
+// Provide the C function to update Field3D by multiplication with Field3D
+void autogen_Field3D_Field3D_multiplication(BoutReal *__restrict__ lhs,
+                                            const BoutReal *__restrict__ rhs, int len) {
+
+  for (int i = 0; i < len; ++i) {
+    lhs[i] *= rhs[i];
+  }
+}
+
+// Provide the C++ operator to update Field3D by multiplication with Field3D
+Field3D &Field3D::operator*=(const Field3D &rhs) {
+  // only if data is unique we update the field
+  // otherwise just call the non-inplace version
+  if (data.unique()) {
+    Indices i{0, 0, 0};
+
+    ASSERT1(fieldmesh == rhs.getMesh());
+
+    checkData(*this);
+    checkData(rhs);
+
+    autogen_Field3D_Field3D_multiplication(&(*this)[i], &rhs[i],
+                                           fieldmesh->LocalNx * fieldmesh->LocalNy *
+                                               fieldmesh->LocalNz);
+
+#if CHECK > 0
+    if (this->getLocation() != rhs.getLocation()) {
+      throw BoutException("Error in multiplication: fields at different locations. lhs "
```

Comment:
Is there anyway to distinguish that this has failed in a `*=` operation rather than the `*` operation?